### DISCUSSION TITLE
controller test -> be_authorized_to -> create? & new?

### DISCUSSION THREAD

**Discussion by danielnc** [palkan/action_policy#214] [Q&A] [answered]
How do I properly test `:create?`

```ruby
    it "is authorized" do
      expect { post_create }.to be_authorized_to(:create?, Product.last)
        .with(Accounts::ProductPolicy)
    end
```

Since block with expect is deferers the post_create, I don't know how to test that I am authorizing the post object to be created

Similar question to :new?

Since Product.new == Product.new is false, I don't know how to implement the same test

**Comment by palkan** [comment, marked as answer]
Doesn't  `an_instance_of(Product)` work here? I believe, it should.

In general, I recommend against calling `authorize!` against non-yet-created records. Creating a record is usually protected by some outer context, which should be authorized.

Check out also this discussion: https://github.com/palkan/action_policy/discussions/209
